### ISSUE TITLE
remove useless hierarchy of Node classes

### ISSUE THREAD

**Issue by michaelficarra** [shapesecurity/shift-spec#34]
Expression, Statement, PrimaryExpression do not add any fields and thus should be removed

**Comment by ikarienator** [comment]
Are you trying to remove reducers from our system, Michael? Don't remove any type that shows up in the type parameter list of `ReducerP`. They are:

```
Script
ProgramBody
ObjectProperty
PropertyName
Identifier
Expression
Directive
Statement
Block
VariableDeclarator
VariableDeclaration
SwitchCase
SwitchDefault
CatchClause
```

You can remove `PrimaryExpression` if you want.

**Comment by michaelficarra** [comment]
Fixed by #39.
